File: sum_n_numbers.py
"""Print the sum of n numbers provided by the user."""


def read_positive_count() -> int:
    """Read and return a positive integer count from stdin."""
    while True:
        raw = input("Enter how many numbers you want to sum: ").strip()
        try:
            count = int(raw)
            if count <= 0:
                print("Please enter a positive integer.")
                continue
            return count
        except ValueError:
            print("Invalid count. Please enter a valid integer.")


def read_number(index: int) -> float:
    """Read and return one numeric value from stdin."""
    while True:
        raw = input(f"Enter number {index}: ").strip()
        try:
            return float(raw)
        except ValueError:
            print("Invalid number. Please enter a valid numeric value.")


def main() -> None:
    n = read_positive_count()
    total = 0.0

    for i in range(1, n + 1):
        total += read_number(i)

    if total.is_integer():
        print(f"Sum: {int(total)}")
    else:
        print(f"Sum: {total}")


if __name__ == "__main__":
    main()

File: sum_n_numbers.cpp
#include <iostream>
#include <limits>

long long sum_upto_n(long long n) {
    return n * (n + 1) / 2;
}

int main() {
    std::cout << "Enter a number: ";

    long long n{};
    if (!(std::cin >> n)) {
        std::cout << "Invalid input\n";
        return 1;
    }

    if (n <= 0) {
        std::cout << "Please enter a positive integer\n";
        return 1;
    }

    __int128 raw_sum = static_cast<__int128>(n) * (n + 1) / 2;
    if (raw_sum > std::numeric_limits<long long>::max()) {
        std::cout << "Number is too large\n";
        return 1;
    }

    std::cout << "Sum: " << sum_upto_n(n) << '\n';
    return 0;
}
